### PR TITLE
LPS-164612 Allow search for UserGroups by userId

### PR DESCRIPTION
**Pull request by RafaelCRC**
update for [LPS-164612](https://issues.liferay.com/browse/LPS-164612)

### CODE REVIEW

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@RafaelCRC these changes look good, but I think this is not quite complete at this stage. At the very least, we need an integration test case to assert that we are indexing the expected values.

I think it would also be good to integrate this into the `UserGroupLocalService` layer so that those searches could also be performed at that level. Then we could also add a test case in `UserGroupLocalServiceTest`. We already have a few test cases for searching defined there.

**Comment by RafaelCRC** [comment]
@drewbrokke updated the code! Thanks!

**Comment by RafaelCRC** [comment]
ci:test:sf

**Comment by RafaelCRC** [comment]
ci:test:relevant

File: modules/apps/user-groups-admin/user-groups-admin-test/src/testIntegration/java/com/liferay/user/groups/admin/service/test/UserGroupLocalServiceTest.java [hunk 1]
```diff
@@ -130,17 +136,25 @@ public void testDatabaseSearchUserUserGroups() throws Exception {
...
-	public void testDatabaseSearchWithInvalidParamKey() throws Exception {
-		List<UserGroup> userGroups = _search(
-			null,
-			LinkedHashMapBuilder.<String, Object>put(
-				UserGroupFinderConstants.PARAM_KEY_USER_GROUPS_ROLES,
-				_role.getRoleId()
-			).put(
-				"invalidParamKey", "invalidParamValue"
-			).build());
+	public void testSearchIndexerDocument() throws Exception {
```
Comment (pei-jung) [inline modules/apps/user-groups-admin/user-groups-admin-test/src/testIntegration/java/com/liferay/user/groups/admin/service/test/UserGroupLocalServiceTest.java:139]: Rename the test to `testSearchUserGroupsWithUserIds` to match with the naming in other `testSearchUserGroups*` tests.

File: modules/apps/user-groups-admin/user-groups-admin-test/src/testIntegration/java/com/liferay/user/groups/admin/service/test/UserGroupLocalServiceTest.java [hunk 2]
```diff
@@ -130,17 +136,25 @@ public void testDatabaseSearchUserUserGroups() throws Exception {
...
+		_userGroupLocalService.addUserUserGroup(user2.getUserId(), userGroup);
+
+		Indexer<UserGroup> indexer = IndexerRegistryUtil.nullSafeGetIndexer(
+			UserGroup.class);
+
+		Document document = indexer.getDocument(userGroup);
+
+		Assert.assertArrayEquals(
+			_getUserGroupUserIds(userGroup),
+			GetterUtil.getLongValues(document.getValues("userIds")));
```
Comment (pei-jung) [inline modules/apps/user-groups-admin/user-groups-admin-test/src/testIntegration/java/com/liferay/user/groups/admin/service/test/UserGroupLocalServiceTest.java:157]: We can utilize the `_search` helper method here for the assertion.

**Review by pei-jung — CHANGES_REQUESTED** [review]

**Comment by melodywu** [comment]
Hello @RafaelCRC , There is a related failure found in ci:test:relevant result, The backend testcase failed [com.liferay.user.groups.admin.service.test.UserGroupLocalServiceTest](https://testray.liferay.com/home/-/testray/case_results/2000339144?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3Fstatuses%3D4%26testrayBuildId%3D2000335610%26testrayRunId%3D2000337313%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc%26status%3D4)
`com.liferay.user.groups.admin.service.test.UserGroupLocalServiceTest > testSearchUserGroupsWithNullParamsAndIndexerDisabled FAILED
     [exec]     com.liferay.portal.kernel.exception.SystemException: com.liferay.portal.kernel.exception.SystemException: java.lang.NullPointerException
     [exec]     	at com.liferay.portal.service.impl.UserGroupLocalServiceImpl.search(UserGroupLocalServiceImpl.java:513)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]     	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:50)
     [exec]     	at com.liferay.portal.spring.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:69)
     [exec]     	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:57)
     [exec]     	at com.liferay.portal.spring.aop.AopInvocationHandler.invoke(AopInvocationHandler.java:49)
     [exec]     	at com.sun.proxy.$Proxy137.search(Unknown Source)` Could you help with confirm? Thanks in advance.

**Comment by RafaelCRC** [comment]
@melodywu Hi! The PR isn't finished yet, it still need some changes requested by Pei-Jung. Sorry about that.

**Comment by RafaelCRC** [comment]
ci:test:sf

**Comment by RafaelCRC** [comment]
ci:test:relevant

**Review by pei-jung — APPROVED** [review]
LGTM!

**Comment by RafaelCRC** [comment]
ci:forward

**Pull request closed without merging**